crates/oidc-client/src/types/client_credentials.rs

```diff
@@ -52,17 +52,17 @@ pub const CLIENT_SUPPORTED_AUTH_METHODS: &[OAuthClientAuthenticationMethod] = &[
 
 /// A function that takes a map of claims and a signing algorithm and returns a
 /// signed JWT.
-pub type JwtSigningFn =
-    dyn Fn(HashMap<String, Value>, JsonWebSignatureAlg) -> Result<String, BoxError> + Send + Sync;
+pub type JwtSigningFn = fn(HashMap<String, Value>, JsonWebSignatureAlg) -> Result<String, BoxError>;
```

Comment:
Doesn't that limit to only static functions, and not closures which capture variables from their environment? Wouldn't that make it harder to use in most cases?

Reply:
Ah right I didn't read the fine prints :sweat_smile:

I don't know if using an `Arc` is too much though. I assume it's either this or making `apply_to_request` use a reference.

Reply:
The `Arc` is fine with me if it works 👍